### PR TITLE
Add GeneratedCodeAttribute to ThisAssembly

### PR DESCRIPTION
**Pull request by JohannesWill**
Add GeneratedCodeAttribute to ThisAssembly

### CODE REVIEW

**Comment by AArnott** [comment]
Thanks for contributing.

Why is this a worthwhile attribute to include? I'm not too familiar with this attribute.

**Comment by JohannesWill** [comment]
The GeneratedCodeAttribute class can be used by code analysis tools to identify computer-generated code, and to provide an analysis based on the tool and the version of the tool that generated the code.

see 
https://docs.microsoft.com/en-us/dotnet/api/system.codedom.compiler.generatedcodeattribute?view=netframework-4.7.2

**Comment by AArnott** [comment]
Interesting. We have several code generators. You've only updated the C# one. We have other languages, and [codedom-based one](https://github.com/AArnott/Nerdbank.GitVersioning/blob/166a3e3bd611732b5f451266a27d9c8185bffbe7/src/Nerdbank.GitVersioning.Tasks/AssemblyVersionInfo.cs#L118). Can you update them all so they're consistent?

File: src/Nerdbank.GitVersioning.Tasks/AssemblyVersionInfo.cs [hunk 1]
```diff
@@ -119,8 +119,14 @@ private CodeTypeDeclaration CreateThisAssemblyClass()
             {
                 IsClass = true,
                 IsPartial = true,
-                TypeAttributes = TypeAttributes.NotPublic | TypeAttributes.Sealed,
+                TypeAttributes = TypeAttributes.NotPublic | TypeAttributes.Sealed | TypeAttributes.Abstract,
```
Comment (AArnott) [inline src/Nerdbank.GitVersioning.Tasks/AssemblyVersionInfo.cs]: Curious: Why was `Abstract` a required addition?
Does it make sense to be both abstract *and* sealed? (perhaps that's the way C# "static" classes are actually implemented?)

Comment (JohannesWill) [reply src/Nerdbank.GitVersioning.Tasks/AssemblyVersionInfo.cs]: It was for the C# "static" classes 
I removed it.

Comment (AArnott) [reply src/Nerdbank.GitVersioning.Tasks/AssemblyVersionInfo.cs]: Thanks.  I didn't know you could create static classes with CodeDOM. Nice trick. :)

**Comment by AArnott** [comment]
Thanks for contributing!

**Pull request merged**